From 1ee0217945cf6e7efda35247de6ba908e4d8c489 Mon Sep 17 00:00:00 2001
From: David Steele <dsteele@gmail.com>
Date: Mon, 7 Oct 2013 21:17:27 -0400
Subject: [PATCH 15/20] Fix for another unhandled exception.

This one would be caught by a previous fix to process_jobs().
This change restores the close() call to the connection.

ERROR: Could not Connect to Cloud Service. Will Try again in %d Seconds
012Traceback (most recent call last):
012  File "/usr/share/cloudprint/cloudprint/cloudprint.py", line 371, in process_jobs
012    XMPP_USE_SSL,xmpp_auth)
012  File "/usr/share/cloudprint/cloudprint/xmpp.py", line 123, in connect
012    self._msg('<auth xmlns="urn:ietf:params:xml:ns:xmpp-sasl" mechanism="X-GOOGLE-TOKEN">%s</auth>' % sasl_token)
012  File "/usr/share/cloudprint/cloudprint/xmpp.py", line 87, in _msg
012    assert not elem.tag.endswith('failure') and not elem.tag.endswith('error')
012AssertionError
(cherry picked from commit 2c3582f87d281d612280def305871f8ce3dd20fc)
---
 cloudprint/xmpp.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cloudprint/xmpp.py b/cloudprint/xmpp.py
index 9667c35..602e7d3 100644
--- a/cloudprint/xmpp.py
+++ b/cloudprint/xmpp.py
@@ -125,7 +125,7 @@ class XmppConnection(object):
             iq = self._msg('<iq type="set"><bind xmlns="urn:ietf:params:xml:ns:xmpp-bind"><resource>Armooo</resource></bind></iq>')
             bare_jid = iq[0][0].text.split('/')[0]
             self._msg('<iq type="set" to="%s"><subscribe xmlns="google:push"><item channel="cloudprint.google.com" from="cloudprint.google.com"/></subscribe></iq>' % bare_jid)
-        except Exception:
+        except:
             self.close()
             raise
 
-- 
1.8.4.rc3

